docs/guides/composing-messages.md

```diff
@@ -69,4 +69,147 @@ ChatPostMessageResponse response = slack.methods(token).chatPostMessage(req -> r
+                value("v1")
+            }
+            button {
+                text("Kin Khao", emoji = true)
+                value("v2")
+            }
+        }
+    }
+}
+```
+
+### Installation
+
+You can add the Block Kit Kotlin DSL via 2 artifacts:
+
+  * The slack model kotlin extension, which adds the Kotlin DSL itself as well as the standalone `withBlocks { }` builder
+  * The slack client kotlin extension, which adds the `.blocks { }` extension function to various Slack Client builders for seamless use of the DSL with the Java builders
+    * The `.blocks { }` extension function is available on the following Java builders:
+      * ChatPostEphemeralRequestBuilder
+      * ChatPostMessageRequestBuilder
+      * ChatScheduleMessageRequestBuilder
+      * ChatUpdateRequestBuilder
+
+**Adding via Gradle:**
+
+```groovy
+dependencies {
+    implementation "com.slack.api:slack-api-model-kotlin-extension:{{ site.sdkLatestVersion }}"
+    implementation "com.slack.api:slack-api-client-kotlin-extension:{{ site.sdkLatestVersion }}"
+}
+```
+
+**Adding via Gradle Kotlin DSL:**
+
+```kotlin
+dependencies {
+    implementation("com.slack.api:slack-api-model-kotlin-extension:{{ site.sdkLatestVersion }}")
+    implementation("com.slack.api:slack-api-client-kotlin-extension:{{ site.sdkLatestVersion }}")
+}
+```
+
+### Notable examples and features
+
+**Standalone withBlocks builder which comes with the model extension**:
+
+```kotlin
+import com.slack.api.model.kotlin_extension.block.withBlocks
+
+val constructedMessage = withBlocks {
+    section {
+        plainText("Now this can be passed to anything that requires a list of LayoutBlocks")
+    }
+}
+```
+
+**Type safe enums for inputs which require specific string inputs**:
+
+```kotlin
+import com.slack.api.model.kotlin_extension.block.element.ButtonStyle
+import com.slack.api.model.kotlin_extension.block.element.ConversationType
+import com.slack.api.model.kotlin_extension.block.withBlocks
+
+val message = withBlocks { 
+    section { 
+        plainText("Please select the person or group you would like to send a cat GIF to.")
+
+        // "accessory" is provided here, but it can be omitted for brevity
+        accessory { 
+            conversationsSelect { 
+                // Or alternatively, provide strings via filter("im", "mpim") if you'd prefer
+                filter(ConversationType.IM, ConversationType.MULTIPARTY_IM)
+                placeholder("Where should we send the cat?")
+                
+                confirm { 
+                    title("Confirm destination")
+                    plainText("Are you sure you want to send a cat GIF to this person or group?")
+                    confirm("Yes, send it")
+                    deny("Don't send it")
+                    
+                    style(ButtonStyle.PRIMARY)
+                }
+            }
+        }
+    }
+}
+```
+
+**Write DSL extension functions for message templating**:
+
```

Comment:
We can express what it is more clearly here. For instance, having a short sentence like "It's even possible to write your own reusable function that can be embedded as part of the Kotlin DSL." at the beginning of this section may be helpful for readers to understand what this part is.